cli/src/cmd/forge/coverage.rs

```diff
@@ -0,0 +1,283 @@
+//! Coverage command
+use crate::{
+    cmd::{
+        forge::{build::CoreBuildArgs, test::Filter},
+        Cmd,
+    },
+    compile::ProjectCompiler,
+    utils,
+};
+use cast::trace::identifier::TraceIdentifier;
+use clap::{AppSettings, ArgEnum, Parser};
+use ethers::{
+    prelude::{Artifact, Project, ProjectCompileOutput},
+    solc::{artifacts::contract::CompactContractBytecode, sourcemap::SourceMap, ArtifactId},
+};
+use forge::{
+    coverage::{
+        CoverageMap, CoverageReporter, DebugReporter, LcovReporter, SummaryReporter, Visitor,
+    },
+    executor::opts::EvmOpts,
+    result::SuiteResult,
+    trace::identifier::LocalTraceIdentifier,
+    MultiContractRunnerBuilder,
+};
+use foundry_common::evm::EvmArgs;
+use foundry_config::{figment::Figment, Config};
+use std::{collections::HashMap, path::PathBuf, sync::mpsc::channel, thread};
+
+// Loads project's figment and merges the build cli arguments into it
+foundry_config::impl_figment_convert!(CoverageArgs, opts, evm_opts);
+
+/// Generate coverage reports for your tests.
+#[derive(Debug, Clone, Parser)]
+#[clap(global_setting = AppSettings::DeriveDisplayOrder)]
+pub struct CoverageArgs {
+    #[clap(
+        long,
+        arg_enum,
+        default_value = "summary",
+        help = "The report type to use for coverage."
+    )]
+    report: CoverageReportKind,
+
+    #[clap(flatten, next_help_heading = "TEST FILTERING")]
+    filter: Filter,
+
+    #[clap(flatten, next_help_heading = "EVM OPTIONS")]
+    evm_opts: EvmArgs,
+
+    #[clap(flatten, next_help_heading = "BUILD OPTIONS")]
+    opts: CoreBuildArgs,
+}
+
+impl CoverageArgs {
+    /// Returns the flattened [`CoreBuildArgs`]
+    pub fn build_args(&self) -> &CoreBuildArgs {
+        &self.opts
+    }
+
+    /// Returns the currently configured [Config] and the extracted [EvmOpts] from that config
+    pub fn config_and_evm_opts(&self) -> eyre::Result<(Config, EvmOpts)> {
+        // Merge all configs
+        let figment: Figment = self.into();
+        let evm_opts = figment.extract()?;
+        let config = Config::from_provider(figment).sanitized();
+
+        Ok((config, evm_opts))
+    }
+}
+
+impl Cmd for CoverageArgs {
+    type Output = ();
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        let (config, evm_opts) = self.configure()?;
+        let (project, output) = self.build(&config)?;
+        println!("Analysing contracts...");
```

Comment:
nit: we now have silent in `CorebuildArgs` so this should print via the `p_println!` macro